---
int32 errorCode
string errorDescription